# HG changeset patch
# User ttl <ttl@justmail.de>
# Date 1312031982 -7200
#      Sat Jul 30 15:19:42 2011 +0200
# Node ID ad18a8e7fb0212b5edc5c523be4f540739b9cb4c
# Parent  0d18b0e168ae3398f1a82208dbba6598817933d0
editor: change window title if text is modified

diff --git a/gui/src/FileEditorMdiSubWindow.cpp b/gui/src/FileEditorMdiSubWindow.cpp
--- a/gui/src/FileEditorMdiSubWindow.cpp
+++ b/gui/src/FileEditorMdiSubWindow.cpp
@@ -47,16 +47,28 @@ FileEditorMdiSubWindow::closeEvent(QClos
     }
   else
     {
       event->accept();
     }
 }
 
 void
+FileEditorMdiSubWindow::handleModificationChanged(bool modified)
+{
+  if ( modified )
+    {
+      QString title(m_fileName);
+      setWindowTitle(title.prepend("* "));
+    }
+  else
+     setWindowTitle (m_fileName);
+}
+
+void
 FileEditorMdiSubWindow::openFile ()
 {
     if (checkFileModified ("Open File")==QMessageBox::Cancel)
       {
         return; // existing file not saved and opening another file canceled by user
       }
     QString openFileName =
         QFileDialog::getOpenFileName (this, "Open File", QDir::homePath(), SAVE_FILE_FILTER);
@@ -334,13 +346,16 @@ FileEditorMdiSubWindow::construct ()
   //       shown directly at the theme icons
   connect (newAction, SIGNAL (hovered ()), this, SLOT (showToolTipNew ()));
   connect (openAction, SIGNAL (hovered ()), this, SLOT (showToolTipOpen ()));
   connect (undoAction, SIGNAL (hovered ()), this, SLOT (showToolTipUndo ()));
   connect (redoAction, SIGNAL (hovered ()), this, SLOT (showToolTipRedo ()));
   connect (saveAction, SIGNAL (hovered ()), this, SLOT (showToolTipSave ()));
   connect (saveAsAction, SIGNAL (hovered ()), this,SLOT (showToolTipSaveAs ()));
 
+  // connect modified signal
+  connect (m_editor, SIGNAL (modificationChanged(bool)), this, SLOT (handleModificationChanged(bool)) );
+
   m_fileName = "";
   setWindowTitle (m_fileName);
   setWindowIcon(QIcon::fromTheme("accessories-text-editor",style->standardIcon (QStyle::SP_FileIcon)));
   show ();
 }
diff --git a/gui/src/FileEditorMdiSubWindow.h b/gui/src/FileEditorMdiSubWindow.h
--- a/gui/src/FileEditorMdiSubWindow.h
+++ b/gui/src/FileEditorMdiSubWindow.h
@@ -47,25 +47,28 @@ Q_OBJECT public:
   void saveFileAs ();
 
   void showToolTipNew ();
   void showToolTipOpen ();
   void showToolTipSave ();
   void showToolTipSaveAs ();
   void showToolTipUndo ();
   void showToolTipRedo ();
-
   void registerModified (bool modified);
 
 protected:
   void closeEvent(QCloseEvent *event);
 
 private:
   int checkFileModified (QString msg);
   void construct ();
   QToolBar *m_toolBar;
   QsciScintilla *m_editor;
   QStatusBar *m_statusBar;
   QString m_fileName;
   bool m_modified;
+
+private slots:
+  void handleModificationChanged(bool modified);
+
 };
 
 #endif // FILEEDITORMDISUBWINDOW_H
